tests/sparsity/test_common.py

```diff
@@ -225,3 +228,30 @@ def test_polynomial_schedule_per_optimizer_step(self, magnitude_algo_mock, speci
             if last_ref_level is not None:
                 mock.assert_called_once_with(pytest.approx(last_ref_level))
             mock.reset_mock()
+
+def test_create_rb_algo_with_per_layer_scheduler_and_per_layer_loss():
+    config = get_empty_config()
+    config['compression'] = {'algorithm': 'rb_sparsity', "params": {"schedule": 'per_layer'}}
+    _, compression_ctrl = create_compressed_model_and_algo_for_test(MockModel(), config)
+
+    # pylint: disable=protected-access
+    assert isinstance(compression_ctrl._loss, PerLayerSparseLoss)
+
+def test_rb_sparsity_per_layer_scheduler__can_set_sparsity_level_for_module():
+    config = get_empty_config()
+    config['compression'] = {'algorithm': 'rb_sparsity', "params": {"schedule": 'per_layer'}}
+    _, compression_ctrl = create_compressed_model_and_algo_for_test(MockModel(), config)
+
+    # pylint: disable=protected-access
+    assert list(compression_ctrl._loss.per_layer_target.values())[0] == 1
+
+    compression_ctrl._scheduler._set_sparsity_level_for_module('MockModel/NNCFLinear[field]', 0.7)
+
+    assert list(compression_ctrl._loss.per_layer_target.values())[0] == pytest.approx(0.3)
```

Comment:
Also, in these kinds of tests you should mock the target `F.linear` function call, check its arguments, count the number of zeros in the sparsified tensor and then compare it against the reference. Should also check that setting a sparsity level for a layer does not impact the other layers. Only after this we will be sure that the feature is functional.